### DISCUSSION TITLE
How to SSG

### DISCUSSION THREAD

**Discussion by yoxud** [antfu/vitesse#423] [General]
Hi, is it possible to give an example of how to generate a series of static pages at build time using this. I've tried initializing the initial state as suggested and populating [product].vue pages  but the [].vue pages are never rendered on build....

**Comment by yoxud** [comment]
So I have a dynamic [productid].vue component. I have removed all dynamic routes (":") in the includedRoutes() export and returned a fixed number of routes for specific product ids.  . This effectively generates the correct static product files upon build. However, when I serve the site, the dynamic routes are still there and overwriting the static ones. How can I  remove the dynamic paths for the client only (not the ssg server) and have it use the generated files?

**Comment by spokospace** [comment]
I have a similar problem, also with dynamic product/category pages. 
Did u resolved this?

It will be good to add example of dynamic page with correct settings of sitemap and ssg to vitesse.

I have structure like this:

[lang]/[category]

and

[lang]/product/[product-id].vue


tried to configure vite-ssg in many ways, but doesn't know completely how to set up custom dynamic paths, couldn't find any existing example
